tools/tnodb_mock/provision.go

```diff
@@ -92,11 +92,10 @@ func getRes(nodeID string, all bool) []*provision.Reservation {
 			continue
 		}
 		// if we are long polling, only return the new reservation
-		if !all && r.Sent {
+		if !all && r.Reservation.Result != nil || r.Reservation.Expired() {
```

Comment:
So i see why u store the result on the reservation. But I feel this is not the optimum way to mark reservations as (processes).

Also don't you need to actually re-create all old provisions on reboot of the node anyway?

Reply:
It doesn't hurt regarding the reboot flow, cause at boot I ask BCDB to send me all the reservation.
But you are right, a single reservation can potentially be deployed multiple time of the node reboots. So maybe we should keep a list of result per reservation. And having it inside the Reservation object is probably also wrong. We can definitely  move it out.